src/storage_proof.rs

```diff
@@ -256,4 +260,13 @@ pub mod tests {
             Err(StorageProofError::DuplicateNodes)
         ));
     }
+
+    #[test]
+    fn invalid_compact_proof_does_not_panic_when_decoding() {
+        let invalid_proof = CompactProof {
+            encoded_nodes: vec![vec![135]],
+        };
+        let result = invalid_proof.to_memory_db::<Hasher>(None);
+        assert!(result.is_ok()); // Should not panic, even with invalid data
```

Comment:
Why would an invalid proof not panic? 

Shouldn't it panic??

Reply:
We don't do any checking of proof validity on the conversion to and from a compact proof anymore.